c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -706,6 +737,17 @@ class DeviceCachingAllocator {
     }
 
     block->allocated = true;
+    if (context) {
+      trimHistoryBefore(block, (char*)block->ptr + size);
+      block->history = std::make_unique<History>(History{
```

Comment:
nit: Not need to create the temporary object here std::make_unique<History>(block->ptr, ...)

Reply:
I thought so too but GCC thinks otherwise... something about {} vs () initialization...